docs/book/v1/intro.md

```diff
@@ -0,0 +1,143 @@
+# Introduction
+
+This library provides the ability to dispatch middleware pipelines in place of
+controllers within laminas-mvc.
+
+## Dispatching PSR-7 Middleware
+
+[PSR-7](http://www.php-fig.org/psr/psr-7/) defines interfaces for HTTP messages,
+and is now being adopted by many frameworks; Laminas itself offers a
+parallel microframework targeting PSR-7 with [Mezzio](https://docs.mezzio.dev/mezzio).
+What if you want to dispatch PSR-7 middleware from [laminas-mvc](https://docs.laminas.dev/laminas-mvc/)?
+
+laminas-mvc currently uses [laminas-http](https://docs.laminas.dev/laminas-http/)
+for its HTTP transport layer, and the objects it defines are not compatible with
+PSR-7, meaning the basic MVC layer does not and cannot make use of PSR-7
+currently.
+
+However, starting with version 2.7.0, laminas-mvc offers
+`Laminas\Mvc\MiddlewareListener`. This `Laminas\Mvc\MvcEvent::EVENT_DISPATCH`
+listener listens prior to the default `DispatchListener`, and executes if the
+route matches contain a "middleware" parameter, and the service that resolves to
+is callable. When those conditions are met, it uses the [PSR-7 bridge](https://docs.laminas.dev/laminas-psr7bridge/)
+to convert the laminas-http request and response objects into PSR-7 instances, and
+then invokes the middleware.
+
+Starting with laminas-mvc version 3.2.0, `Laminas\Mvc\MiddlewareListener` is deprecated and replaced
+by `Laminas\Mvc\Middleware\MiddlewareListener` provided by this package.  
+After package installation, `Laminas\Mvc\Middleware` module must be registered in your
+laminas-mvc based application.
+
+## Mapping Routes to Middleware
+
+The first step is to map a route to PSR-7 middleware. This looks like any other
+[routing](https://docs.laminas.dev/laminas-mvc/routing/) configuration,
+with one small change: instead of providing a `controller` in the routing
+defaults, you provide `middleware`:
+
+```php
+// Via configuration:
+use Application\Middleware\IndexMiddleware;
+use Laminas\Router\Http\Literal;
+
+return [
+    'router' => [
+        'routes' => [
+            'home' => [
+                'type' => Literal::class,
+                'options' => [
+                    'route' => '/',
+                    'defaults' => [
+                        'middleware' => IndexMiddleware::class,
+                    ],
```

Comment:
This example should have `controller` set to PipeSpec
```suggestion
                    'defaults' => [
                        'controller' => PipeSpec::class,
                        'middleware' => IndexMiddleware::class,
                    ],
```

Reply:
No, it's the documentation of version 1.

Reply:
then why is it new?! it was a trap all along

Reply:
> then why is it new?! it was a trap all along

In the `1.x` branch the docs don't have a `v1` directory. The `intro` file there was changed to a redirect. It's not a trap, it's a feature 🙂